### ISSUE TITLE
dynamically disable non-applicable filter items in notifications view

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#4846]
### Observed behavior

reported by @jtamiace  during an 0.12.0 alpha hack session:

![image](https://user-images.githubusercontent.com/2367265/52078055-22e74b80-2547-11e9-83a8-1dd0a559ef7d.png)

### Expected behavior

@jtamiace correct me if I'm wrong but it sounds like the behavior you expect would be:

> Filter items in both drop-downs (resource type, progress type) should only be enabled if selecting them would have greater than 0 results

### User-facing consequences

extra options that don't do anything

### Steps to reproduce

filter notifications such that either the first or second filter 

### Context

0.12.0 alpha 7

**Comment by jtamiace** [comment]
Yes that would solve this issue since help will never be needed for non-exercises, and will also cover the other cases where the filter would return empty

**Comment by indirectlylit** [comment]
@jonboiser I think you fixed this, correct?

**Comment by jonboiser** [comment]
It's partially fixed, but pretty naïve. It just looks to see what set of resources or statuses are represented in the list, but does not smartly update the filters as they are applied. E.g. if you have a completed lesson and a started video, you will get all of the filters: completed, started, lesson, and video, even if you don't have a completed video.

**Comment by indirectlylit** [comment]
okay cool – will leave this open for future improvements

**Comment by Pursottam6003** [comment]
Hi @marcellamaki  I also would like to work on this issue if its possible please assign me this issue

**Comment by MisRob** [comment]
Hello @Pursottam6003, thank you for your interest. As you volunteered for more issues and already have some of them assigned, I'm leaving this unassigned for now in case someone else wanted to tackle it as you're working on other issues. When your plate is empty, if this issue is still available, please remind yourself here and we can assign it to you.
